typescript/ecs/capacity-providers/index.ts

```diff
@@ -0,0 +1,166 @@
+ * Case 2: m services placed on spot only (100% spot)
+ * Case 3: n services first placed on spot, remaining m-n placed on on-demand
+ * Case 4: n services first placed on on-demand, remaining m-n placed on spot
+ *
+ * The case of all m services being placed on on-demand capacity is equivalent to using the fargate launchType without
+ * capacity providers at all, so will not be considered in this example.
+ */
+
+ // Case 1: m services split across both spot and on-demand
+ 
+const service1 = new ecs.FargateService(stack, 'FargateService-NoBase-EqualWeight', {
+  cluster,
+  taskDefinition,
+  desiredCount: 10,
+  capacityProviderStrategies: [
+    {
+      capacityProvider: 'FARGATE_SPOT',
+      weight: 1,
+    },
+    {
+      capacityProvider: 'FARGATE',
+      weight: 1,
+    },
+  ],
+});
+
+ //  Case 2: Service with Spot weight only
+const service2 = new ecs.FargateService(stack, 'FargateService-AllSpot', {
+  cluster,
+  taskDefinition,
+  desiredCount: 10,
+  capacityProviderStrategies: [
+    {
+      capacityProvider: 'FARGATE_SPOT',
+      weight: 1,
+    },
+    {
+      capacityProvider: 'FARGATE',
+      weight: 0,
+    },
+  ],
+});
+
+/*
+ * Case 3: n services first placed on spot, remaining m-n placed on on-demand
+ * Here, m = 5 and n = 2
+ */
+const service3 = new ecs.FargateService(stack, 'FargateService-Spot-Base', {
+  cluster,
+  taskDefinition,
+  desiredCount: 5,
+  capacityProviderStrategies: [
+    {
+      capacityProvider: 'FARGATE_SPOT',
+      base: 2,
+      weight: 0,
+    },
+    {
+      capacityProvider: 'FARGATE',
+      weight: 1,
+    },
+  ],
+});
+
+/*
+ * Case 4: n services first placed on on-demand, remaining m-n placed on spot
+ * Here, m = 5 and n = 2
+ */
+
+const service4 = new ecs.FargateService(stack, 'FargateService-OnDemand-Base', {
+  cluster,
+  taskDefinition,
+  desiredCount: 5,
+  capacityProviderStrategies: [
+    {
+      capacityProvider: 'FARGATE',
+      base: 2, // dedicated
+      weight: 0,
+    },
+    {
+      capacityProvider: 'FARGATE_SPOT',
+      weight: 1,
+    },
+  ],
+});
+
+// Create ALB
+const lb = new elb.ApplicationLoadBalancer(stack, 'LB', {
+  vpc,
+  internetFacing: true
+});
+const listener = lb.addListener('PublicListener', { port: 80, open: true });
+
+// Attach ALB to ECS Service
+listener.addTargets('ECS', {
+  port: 80,
+  // never actually do this
+  targets: [
+    service1.loadBalancerTarget({
+      containerName: 'web',
+      containerPort: 80
+    }),
+    service2.loadBalancerTarget({
+      containerName: 'web',
+      containerPort: 80
+    }),
+    service3.loadBalancerTarget({
+      containerName: 'web',
+      containerPort: 80
+    }),
+    service4.loadBalancerTarget({
+      containerName: 'web',
+      containerPort: 80
+    }),
+  ],
+  // include health check (default is none)
+  healthCheck: {
+    interval: cdk.Duration.seconds(60),
+    path: "/health",
+    timeout: cdk.Duration.seconds(5),
+  }
+});
```

Comment:
same do we need to add this extra LB section, seems like the services are fine on their own.